src/c++/perf_analyzer/genai-perf/genai_perf/parser.py

```diff
@@ -280,7 +287,7 @@ def _add_endpoint_args(parser):
         "-m",
         "--model",
         type=str,
-        required=True,
+        default=None,
```

Comment:
This would break the profile workflow if you did not add the check_model_args method?
Am I understanding that correctly?

Reply:
I want to make sure we did not break any of the unit testing. Is this currently passing?

Reply:
This is mainly to make compare subcommand work. If this is required, then we cannot do `genai-perf compare ...` because we are missing `-m` option (we will need to call it like `genai-perf -m <model> compare ...`). 

The model name is required because we need model names to prep for PA run and that is being done under _check_model_args.

Reply:
I think this leads naturally to making profile a sub command.

Reply:
Are you going to do that as a separate ticket?

Reply:
Do you mean adding a profile subcommand?

Reply:
I agree. It sounds like that is an active discussion, but we're going to run into these types of issues if we don't make each different use of GenAI-Perf a subcommand. I think if we're going to start coding things to work around or duplicate what our libraries do by default (e.g. the required check), it usually means we're doing something wrong design-wise.

Hyunjae, if we do end up making a separate sub-command and you are the person to implement it, please make sure to create a ticket for that extra work so that your time is being tracked. You should also update the relevant design document, since it sounds like this was out of scope for it.